### PR TITLE
Add discount code lookup

### PR DESCRIPTION
**Pull request by davidlawson**
This is a potential fix for #403, adding a lookup method to DiscountCode.

Usage:
```
code = ShopifyAPI::DiscountCode.lookup('...')
```

### CODE REVIEW

**Comment by jamiemtdwyer** [comment]
It's definitely not pretty... but this is probably the only way to solve this in this library.

File: lib/shopify_api/resources/discount_code.rb [hunk 1]
```diff
@@ -5,5 +5,15 @@ class DiscountCode < Base
     def price_rule_id
       @prefix_options[:price_rule_id]
     end
+
+    def self.lookup(code)
+      find(:one, from: '/admin/discount_codes/lookup.json?code=' + code)
+    rescue ActiveResource::Redirection => ex
+      r = find(:one, from: ex.response['Location'])
```
Comment (jamiemtdwyer) [inline lib/shopify_api/resources/discount_code.rb]: if we store `ex.response['Location']` to a variable, there will be a little less repetition here

**Comment by tnastine** [comment]
I created a price rule/discount code ("1234") to a development shop and have been trying to get this to work simply by triggering the call on the Home Controller index function. Trying to display the price_rule_id that it finds for the specific code to the Index View.


My API call from Home Controller:

`@price_rule_id = ShopifyAPI::DiscountCode.lookup("1234")`

This is what the view is displaying:

"<ShopifyAPI::DiscountCode:0x007d19686a32f0>"


I'm relatively new to Ruby -- am I missing something?

File: lib/shopify_api/resources/discount_code.rb (hunk 1 above)
Comment (davidlawson) [reply lib/shopify_api/resources/discount_code.rb]: good idea, made the change.

**Comment by davidlawson** [comment]
@tnastine try 
```
c = ShopifyAPI::DiscountCode.lookup("1234")
puts c.attributes
```
to see the attributes, then you can do e.g. `c.id` to get the ID

**Comment by tnastine** [comment]
And of course I figure it out two minutes later. Sorry for the clutter.

I didn't include `.prefix[:price_rule_id]` on my variable in the view. Working now 100%.

**Comment by nwtn** [comment]
Looks like this dropped off everyone’s radar, but it seems like it’s worth adding. @davidlawson are you able to add a test for this method?

**Comment by Nkadze** [comment]
really useful pull, is it planned to merge on master?

**Comment by dan003400** [comment]
Why is this not being merged in?

**Pull request closed without merging**